### PR TITLE
Develop CLI training examples

### PR DESCRIPTION
**Pull request by jeromelecoq**
The goal of this PR is to have a series of examples to facilitate adoption of the CLI  for training. We will develop inference examples in a separate PR.

### CODE REVIEW

File: examples/cli_example_tiny_ephys_training.py [hunk 1]
```diff
@@ -61,41 +53,28 @@
...
 
 training_param["model_string"] = (
     network_param["name"]
     + "_"
     + training_param["loss"]
-    + "_"
-    + run_uid
 )
-training_param["output_dir"] = "/Users/jeromel/Desktop/tmp"
+training_param["output_dir"] = str(pathlib.Path(__file__).parent.absolute())
```
Comment (danielsf) [inline examples/cli_example_tiny_ephys_training.py:69]: Could you wrap all of this code in a `__main__` block and use `argparse` to allow the user to specify the output? I get that this is just an example, but I'm a bit leery of public-facing code that automatically chooses an output directory without user input.

File: examples/cli_example_tiny_ophys_training.py [hunk 2]
```diff
@@ -61,41 +53,28 @@
...
 network_param["name"] = "unet_single_1024"
 # Name of network topology in the collection
 
 # Those are parameters used for the training process
-training_param["type"] = "trainer"
 training_param["name"] = "core_trainer"
 training_param["learning_rate"] = 0.0001
 training_param["loss"] = "mean_squared_error"
-training_param["output_dir"] = "/Users/jeromel/Desktop/tmp"
+training_param["output_dir"] = str(pathlib.Path(__file__).parent.absolute())
```
Comment (danielsf) [inline examples/cli_example_tiny_ophys_training.py:63]: Same comment about using `argparse` to let the user specify the output.

I suppose an alternative would be to add a print statement informing the user that the script was about to write output to this directory.

File: examples/cli_example_training_ephys.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+  "generator_params": {
+    "pre_post_frame": 30,
+    "steps_per_epoch": 10,
+    "total_samples": -1,
+    "name": "EphysGenerator",
+    "batch_size": 100,
+    "start_frame": 2000,
+    "randomize": 0,
+    "train_path": "/Users/jeromel/Documents/Work documents/Allen Institute/Projects/Deep2P/repos/public/deepinterpolation/sample_data/ephys_tiny_continuous.dat2",
```
Comment (danielsf) [inline examples/cli_example_training_ephys.json:10]: I assume the point of this file is to show users what the JSON file needs to look like (in which case, it probably doesn't matter that this directory is on your personal machine)?

File: examples/cli_example_training_ophys.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{"training_params": {"name": "core_trainer", "learning_rate": 0.0001, "loss": "mean_squared_error", "output_dir": "/Users/jeromel/Documents/Work documents/Allen Institute/Projects/Deep2P/repos/public/deepinterpolation/examples", "model_string": "unet_single_1024_mean_squared_error"}, "generator_params": {"steps_per_epoch": 5, "name": "SingleTifGenerator", "pre_post_frame": 30, "train_path": "/Users/jeromel/Documents/Work documents/Allen Institute/Projects/Deep2P/repos/public/deepinterpolation/examples/../sample_data/ophys_tiny_761605196.tif", "batch_size": 5, "start_frame": 0, "end_frame": 100, "pre_post_omission": 0}, "test_generator_params": {"name": "SingleTifGenerator", "pre_post_frame": 30, "train_path": "/Users/jeromel/Documents/Work documents/Allen Institute/Projects/Deep2P/repos/public/deepinterpolation/examples/../sample_data/ophys_tiny_761605196.tif", "batch_size": 5, "start_frame": 0, "end_frame": 100, "pre_post_omission": 1, "steps_per_epoch": -1}, "network_params": {"name": "unet_single_1024"}, "output_full_args": true}
```
Comment (danielsf) [inline examples/cli_example_training_ophys.json:1]: Can you rewrite this with `json.dumps(...., indent=2)` so that it is more human readable?

**Review by danielsf — APPROVED** [review]
The only comment I feel strongly about is the reformatting of `examples/cli_example_training_ophys.json` so that it is easier for humans to read. Everything else should be treated as a mild suggestion.

File: examples/cli_example_training_ephys.json (hunk 3 above)
Comment (jeromelecoq) [reply examples/cli_example_training_ephys.json:10]: Yes this is the intent

File: examples/cli_example_training_ophys.json (hunk 4 above)
Comment (jeromelecoq) [reply examples/cli_example_training_ophys.json]: Done. Good suggestion

File: examples/cli_example_tiny_ophys_training.py (hunk 2 above)
Comment (jeromelecoq) [reply examples/cli_example_tiny_ophys_training.py:63]: I am going to pass on this as I would prefer folks to use the CLI than this.

**Pull request merged**